core/src/main/scala/org/apache/spark/deploy/SparkHadoopUtil.scala

```diff
@@ -122,6 +126,14 @@ class SparkHadoopUtil extends Logging {
   }
 
   /**
+   * Schedule a login from the keytab and principal set using the --principal and --keytab
+   * arguments to spark-submit. This login happens only when the credentials of the current user
+   * are about to expire. This method reads SPARK_PRINCIPAL and SPARK_KEYTAB from the environment
+   * to do the login. This method is a no-op in non-YARN mode.
+   */
+  private[spark] def scheduleLoginFromKeytab(): Unit = {}
```

Comment:
Do you need this method in this class? Seems like it's only ever useful if you're using `YarnSparkHadoopUtil`, so might be better to just declare it there.

The scaladoc also doesn't match this implementation, but the actual implementation in the other class.